### ISSUE TITLE
[1] Setup Pull-Request template

### ISSUE THREAD

**Issue by gibbsfromncis** [bitshares/bitshares-ui#1939]
- [ ] Collect criteria & requirements for Pull-Requests template
- [ ] Create Template

Requested estimate: 1h

**Comment by gibbsfromncis** [comment]
_I like to see the following part of PR template:_

**Description**

**Tested on the following browsers**:

- [ ] Chrome 
- [ ] Opera
- [ ] Firefox
- [ ] Safari

**Screenshots/Licecaps**:

**Comment by gibbsfromncis** [comment]
@svk31 @sschiessl-bcp I would like to remember you that I'm not a guy who do reviews so I maybe do not know what exactly you want to see in PR description. It's really important to hear your thoughts to know how to make life easy for PR reviewers.

**Comment by startailcoon** [comment]
Title should contain issue(s) id and a short description.

Title: `Fixes #xxxx, #xxxx - Adds usefull fix`

Description should do the same and add more details on each issue(s) fix. 

 We could also add checks like:

- [ ] Verified theme compatibility

**Comment by cogutvalera** [comment]
Next points just are my vision, hope/maybe it will help.
Key fields IMHO are:

 1.  PR number
 2.  PR title
 3.  PR description
 4.  PR date
 5.  PR issue(s) [link(s) to another Issue(s) template]
 6.  PR status
 7.  PR estimation
 8.  PR deadline
 9.  PR test (chrome, opera, firefox, safari, ...)
10. PR discussion/comments [link to template]

Just my thoughts ...
Thanks !

**Comment by startailcoon** [comment]
Estimation and deadline is not part of the PR, it's part of the issue.

**Comment by cogutvalera** [comment]
Sorry. You are absolutely right ! Thanks !

**Comment by sschiessl-bcp** [comment]
> Title: Fixes #xxxx, #xxxx - Adds usefull fix
> Description should do the same and add more details on each issue(s) fix.

Agreed. Ideally one PR <> one Issue.

> Please make sure you tested your feature in all following browsers

This can be a problem, as not all devs have access to all browsers (me included). We could ask for Chrome and Firefox test, as the most common browsers. Ideally, this would also be covered with an integration test that we should do for every release. I will tend to #1637 on that matter hopefully soon.

> Screenshots/Licecaps

This should be formulated as only applicable for UI/UX issues. Maybe we can have two templates (since people are too lazy to delete things from the template)? For example, "Code" and "Code + UI".

Can change the template/browser test into a checklist with sections, and add a new section for general things. Add "Up-to-date with latest develop".

**Comment by startailcoon** [comment]
> Agreed. Ideally one PR <> one Issue.

True, but if it covers more than one, due to them being small for example.

> This can be a problem, as not all devs have access to all browsers (me included). We could ask for Chrome and Firefox test, as the most common browsers. Ideally, this would also be covered with an integration test that we should do for every release. I will tend to #1637 on that matter hopefully soon.

The more the merrier. We should still have all the options tickable, to make us aware of how it's been tested. Others can then test those that are missing. 

> Screenshots/Licecaps

It's up to the submitter if there are any to show. Not all in the form can/should be requirements. 

I would like to add a `Tested on Mobile` as well.

**Comment by gibbsfromncis** [comment]
@startailcoon I cannot launch wallet on my iPhone at all so ... maybe we need to fix it before start to require mobile testing?

**Comment by sschiessl-bcp** [comment]
> @startailcoon I cannot launch wallet on my iPhone at all so ... maybe we need to fix it before start to require mobile testing?

+1

**Comment by startailcoon** [comment]
> maybe we need to fix it before start to require mobile testing?

True, having it work on all mobile devices would be nice for that 👍 
Still doesn't mean that all the tick-boxes would be mandatory, but rather a list of items that have been tested. Since you're having issues with it on iPhone you can't tick that box, and it would be weird that you can't verify it, so not having it is fine with me.

**Comment by startailcoon** [comment]
Is there progress on this?
As it's a template issues I guess it doesn't need a specific milestone.

**Comment by sschiessl-bcp** [comment]
Apologies for the delay here ...

**Comment by gibbsfromncis** [comment]
@sschiessl-bcp do you see PR template when try to create an issue? I can't see it...

<img width="1161" alt="screenshot 2019-01-10 03 34 40" src="https://user-images.githubusercontent.com/35899973/50942351-abaf2200-1488-11e9-8b51-8a388d5f6856.png">

**Comment by sschiessl-bcp** [comment]
The way I did it only worked with `?template=...` query parameter,  I thought we get those fancy select buttons as with issues ...

So I switched to having only one template, which is now visible
